lib/board.rb

```diff
@@ -29,9 +29,38 @@ def valid_coordinate?(coordinate)
   def valid_placement?(ship, coordinates)
     return false unless ship.length == coordinates.length
 
-    coordinates.each_cons(ship.length) do |coordinate|
+    board_letters = []
+    board_numbers = []
+    user_coordinate_letters = []
+    user_coordinate_numbers = []
+    @cells.each_cons(ship.length) do |coordinate|
+      coordinate.each do |cell|
+        board_letters << cell[0][0] 
+        board_numbers << cell[0][1] 
+      end
+      board_letters
+      board_numbers
+      board_letters.uniq.count == 1 || board_numbers.uniq.count == 1
+      coordinates.each do |user_coordinate|
+        user_coordinate_letters << user_coordinate[0]
+        user_coordinate_numbers << user_coordinate[1]
+      end
+      user_coordinate_letters
+      user_coordinate_numbers
```

Comment:
We might be able to replace 48 & 49 with another if that checks that order matters i.e. [A1, A2, A3] works and [A3, A2, A1] shouldn't work.